presto-main/src/main/java/com/facebook/presto/sql/planner/plan/SemiJoinNode.java

```diff
@@ -103,6 +114,18 @@ public Symbol getSemiJoinOutput()
         return filteringSourceHashSymbol;
     }
 
+    @JsonProperty("method")
+    public Optional<Method> getMethod()
+    {
+        return method;
+    }
+
+    public boolean isDistributed()
+    {
+        checkState(method.isPresent(), "method not yet set");
```

Comment:
yes.  join nodes get created without any type, and the type gets set by EnforceRequiredJoinMethods.  But by the time you want to use the information (AddExchanges), it needs to be set.